app/src/main/java/hr/dice/filipbionda/tmdbpractice/ui/homescreen/HomeScreen.kt

```diff
@@ -0,0 +1,562 @@
+                        border =
+                        BorderStroke(
+                            width = dimensionResource(R.dimen.home_screen_headline_image_border_width),
+                            color = MaterialTheme.colorScheme.onSurface
+                        ),
+                        shape = CircleShape,
+                    ),
+            ) {
+                Image(
+                    painter = painterResource(R.drawable.profile_picture),
+                    contentDescription = stringResource(R.string.profile_picture_content_description),
+                    contentScale = ContentScale.FillBounds,
+                )
+            }
+        }
+}
+
+@SuppressLint("UseOfNonLambdaOffsetOverload")
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun HomeScreenSearchBar(modifier: Modifier = Modifier) {
+    var query by rememberSaveable { mutableStateOf("") }
+    var expanded by rememberSaveable { mutableStateOf(false) }
+    val suggestions =
+        rememberSaveable {
+            mutableListOf(
+                "Spider-man 2",
+                "Batman",
+            )
+        }
+
+    val animatedDp by animateDpAsState(
+        targetValue = if (expanded) 0.dp else dimensionResource(R.dimen.home_screen_search_horizontal_padding),
+        animationSpec =
+        tween(
+            durationMillis = 400,
+            easing = FastOutLinearInEasing,
+        ),
+        label = "",
+    )
+
+    val animatedYaxisOffset by animateDpAsState(
+        targetValue = if (expanded) 0.dp else dimensionResource(R.dimen.home_screen_search_bar_y_offset),
+        animationSpec =
+        tween(
+            durationMillis = 400,
+            easing = FastOutLinearInEasing,
+        ),
+        label = "",
+    )
+
+    Box(
+        modifier = modifier.fillMaxSize(),
+    ) {
+        SearchBar(
+            colors =
+            SearchBarDefaults.colors(
+                containerColor = purple_36,
+            ),
+            modifier =
+            Modifier
+                .fillMaxWidth()
+                .height(dimensionResource(R.dimen.home_screen_search_bar_height))
+                .offset(y = animatedYaxisOffset)
+                .padding(horizontal = animatedDp),
+            inputField = {
+                SearchBarDefaults.InputField(
+                    colors =
+                    TextFieldDefaults.colors(
+                        focusedTextColor = white,
+                    ),
+                    query = query,
+                    onSearch = { suggestion ->
+                        expanded = false
+                        query = ""
+                        if (suggestion.isNotEmpty()) {
+                            suggestions.add(suggestion)
+                        }
+                    },
+                    onQueryChange = {
+                        query = it
+                    },
+                    expanded = expanded,
+                    onExpandedChange = {
```

Comment:
Why do you need this

Reply:
It is required parameter in SearchBar composable